packages/language-server/test/snapshots/xml-view-diagnostics/snapshots-utils.ts

```diff
@@ -1,16 +1,36 @@
 import { resolve, sep } from "path";
-import { Diagnostic } from "vscode-languageserver-types";
+import { Diagnostic, Range } from "vscode-languageserver-types";
 import { TextDocument } from "vscode-languageserver";
 import { readJsonSync, readFileSync } from "fs-extra";
 import { expect } from "chai";
-import { sortBy, forEachRight } from "lodash";
+import { sortBy, forEachRight, map } from "lodash";
 import { generateModel } from "@ui5-language-assistant/test-utils";
 import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
 import { getXMLViewDiagnostics } from "../../../src/xml-view-diagnostics";
 
 export const INPUT_FILE_NAME = "input.xml";
 export const OUTPUT_LSP_RESPONSE_FILE_NAME = "output-lsp-response.json";
-export const OUTPUT_RANGES_FILE_NAME = "output-ranges.xml";
+
+export async function snapshotTestLSPDiagnostic(
+  testDir: string
+): Promise<void> {
+  const expectedDiagnostics = getExpectedDiagnosticsLSPResponse(testDir);
+  const actualDiagnostics = await getActualDiagnosticLSPResponse(testDir);
+  expect(actualDiagnostics, "Mismatch LSP Diagnostics response").to.deep.equal(
+    expectedDiagnostics
+  );
+
+  const expectedRanges = map(expectedDiagnostics, (_) => _.range);
+  const expectedXMLWithRangeMarkers = getExpectedXMLWithRIssueRangesMarked(
+    testDir,
+    expectedRanges
+  );
+  const actualXMLWithRangeMarkers = readInputXMLSnippet(testDir, false);
```

Comment:
This is the input file content - it's more "expected" than "actual". This is used to check that the ranges in the input xml and the ranges specified in the expected response are synchronized.
If we don't compare them to the actual ranges at any point, why do we need them in the input xml at all? Or, a better question would be, what do we expect to do with the ranges marked in the XML?
What I wanted was to specify them only in the xml, so that we can reference them in the response without actually writing the line and character numbers, which will make changing the expected response unnecessary when we change the input xml without changing the actual issue (even if its position changes).
What it seems you want to do is use it for better error messages. For this we don't actually need the ranges in the input file since we add the actual and expected ranges according to the LSP response.

Reply:
> what do we expect to do with the ranges marked in the XML?

This is be used for reading the test easily. Since the test only passes if the json and xml are synced, we know that the ranges in the json file must be the same as in the xml, so we can see visually where the diagnostics will appear in the editor.